test/request_reply_message_publisher_test.go

```diff
@@ -130,4 +134,421 @@ var _ = Describe("RequestReplyPublisher", func() {
+			largeByteArray := make([]byte, 16384)
+			timeOut := 5 * time.Second
+
+			// A handler for the request-reply publisher
+			publisherReplyHandler := func(message message.InboundMessage, userContext interface{}, err error) {
+				if err == nil { // Good, a reply was received
+					Expect(message).ToNot(BeNil())
+					payload, _ := message.GetPayloadAsString()
+					fmt.Printf("The reply inbound payload: %s\n", payload)
+				} else if terr, ok := err.(*solace.TimeoutError); ok { // Not good, a timeout occurred and no reply was received
+					// message should be nil
+					// This handles the situation that the requester application did not receive a reply for the published message within the specified timeout.
+					// This would be a good location for implementing resiliency or retry mechanisms.
+					fmt.Printf("The message reply timed out with %s\n", terr)
+				} else { // async error occurred.
+					Expect(err).ToNot(BeNil())
+				}
+			}
+
+			// A helper function to saturate a given publisher. Counts the number of published messages at the given int pointer
+			// Returns a channel that is closed when the publisher receives an error from a call to Publish
+			// Returns a channel that can be closed when the test completes, ie. if an error occurred
+			publisherSaturation := func(publisher solace.RequestReplyMessagePublisher, publishedMessages *int) (publisherComplete, testComplete chan struct{}) {
+				publisherComplete = make(chan struct{})
+				testComplete = make(chan struct{})
+
+				toPublish, err := messageBuilder.BuildWithByteArrayPayload(largeByteArray)
+				Expect(err).ToNot(HaveOccurred())
+
+				go func() {
+					defer GinkgoRecover()
+				loop:
+					for {
+						select {
+						case <-testComplete:
+							break loop
+						default:
+						}
+						err := publisher.Publish(toPublish, publisherReplyHandler, publishTopic, timeOut, nil /* properties */, nil /* usercontext */)
+
+						if err != nil {
+							Expect(err).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+							break loop
+						} else {
+							(*publishedMessages)++
+						}
+					}
+					close(publisherComplete)
+				}()
+				// allow the goroutine above to saturate the publisher
+				time.Sleep(100 * time.Millisecond)
+				return publisherComplete, testComplete
+			}
+
+			It("should publish all messages on graceful termination (no waiting for reply messages)", func() {
+				publishedMessages := 0
+
+				bufferSize := uint(1000)
+				publisher, err := messagingService.RequestReply().CreateRequestReplyMessagePublisherBuilder().OnBackPressureWait(bufferSize).Build()
+				Expect(err).ToNot(HaveOccurred())
+
+				err = publisher.Start()
+				Expect(err).ToNot(HaveOccurred())
+				defer publisher.Terminate(0)
+
+				publisherComplete, testComplete := publisherSaturation(publisher, &publishedMessages)
+				defer close(testComplete)
+
+				// allow the goroutine above to saturate the publisher
+				time.Sleep(100 * time.Millisecond)
```

Comment:
publisherSaturation already sleeps for 100 miliseconds why sleep again? Is there a better way to to use:
```
select{
case <-time.After(100 * time.Milisecond):
// waits for 100 miliseconds
// depending on test this is ok or an error
case <-publisherComplete:
// waits until publisher is done
// depending on test this is ok or an error
}
```